Automattic/jetpack
# Instant Search: Implement minimal search results and spelling correction

## Changes proposed in this Pull Request:
Implementing the minimal search results mockup from p3QzjZ-VL-p2

There will be three different types of search results:
- minimal
- product
- engagement

But this is just focusing on the first one and setting us up for the other two.

This PR also implements the logic for corrected search queries when we do any spelling correction and displaying the total number of results. Here is what it all looks like:

<img width="1250" alt="Screen Shot 2019-08-26 at 8 24 06 PM" src="https://user-images.githubusercontent.com/820871/63736391-1bb7c000-c840-11e9-8c0d-d2705ca73669.png">

Instant Search remains gated by a `JETPACK_SEARCH_PROTOTYPE` define.

## Is this a new feature or does it add/remove features to an existing part of Jetpack?
* Yes, this is a new feature being added onto Jetpack Instant Search (p3QzjZ-V3-p2).

## Testing instructions:
* Add `define( "JETPACK_SEARCH_PROTOTYPE", true );` to your wp-config.php.
* Ensure that your site has the Jetpack Pro plan and has Jetpack Search enabled.
* Enter a query into a search widget. Alternatively, navigate to a search page like `/?s=privacy`.
* Verify how the page looks and that you are getting results as expected 

You can also follow the instructions in #13290 to run your query against any site rather than the site you deploy the code on (all the data comes from the API).

## Proposed changelog entry for your changes:
* No changelog entry necessary; these changes will not be accessible to our users.
